overlord/servicestate/quota_control.go

```diff
@@ -42,6 +50,54 @@ func checkSystemdVersion() {
 
 func init() {
 	checkSystemdVersion()
+	checkMemoryCGroupEnabled()
+}
+
+func checkMemoryCGroupEnabled() {
+	memoryCGroupError = memoryCGroupEnabled()
+}
+
+var CGroupsFilePath = "/proc/cgroups"
+
+//	since the control groups can be enabled/disabled without the kernel config the only
+//	way to identify the status of memory control groups is via /proc/cgroups
+//	"cat /proc/cgroups | grep memory" returns the active status of memory control group
+//	and the 3rd parameter is the status
+//	0 => false => disabled 1 => true => enabled
+func memoryCGroupEnabled() error {
+	cgroupsFile, err := os.Open(CGroupsFilePath)
+	if err != nil {
+		return fmt.Errorf("cannot open cgroups file: %v", err)
+	}
+	defer cgroupsFile.Close()
+	scanner := bufio.NewScanner(cgroupsFile)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "memory\t") {
+			memoryCgroupValues := strings.Fields(line)
+			if len(memoryCgroupValues) == 4 { // any change in size should lead to an error for us to update
```

Comment:
Not sure about the possible placement, since we don't have a key to determine the status I think it might be a good idea to simply fail the test so we can manually confirm that it's still working or update accordingly?

Reply:
I think it's fine to only fail if the size is less than 4 for the reasons @MiguelPires laid out